Move process "convertGFFtoGTF" before "makeSTARindex"

silviamorins opened the pull request:
Running the command:
```
nextflow run nf-core/rnaseq -r 1.3 -profile cfc --fasta 'Streptomyces_coelicolor_a3_2_.ASM20383v1.dna_rm.toplevel.fa' --gff 'Streptomyces_coelicolor_a3_2_.ASM20383v1.43.chromosome.Chromosome.gff' --reads '[...]*_{R1,R2}.fastq.gz' -resume --skip_preseq --reverse_stranded --skip_genebody_coverage
```
I get the following error:
`ERROR ~ No such variable: gtf_makeSTARindex`

As suggested by @lpantano [here](https://github.com/nf-core/rnaseq/issues/215) I tried moving the process `convertGFFtoGTF` before `makeSTARindex`, so that as this last one starts, the necessary input file `${gff.baseName}.gtf` (coming from `convertGFFtoGTF`) exists; the error doesn't show up anymore.

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] Make sure your code lints (`nf-core lint .`).
 - [x] `CHANGELOG.md` is updated

**Learn more about contributing:** https://github.com/nf-core/rnaseq/tree/master/.github/CONTRIBUTING.md

drpatelh commented:
Nice! Great debugging @lpantano ! Thanks @silviamorins :+1:

drpatelh commented:
Did you change `environment.yml` @silviamorins . There seems to be something funky going on with the PR. The `featureCounts` commit isnt showing up either...

silviamorins commented:
I took away a change I had previously pushed to my fork (I had added a `-M` flag) in the `featureCounts` command, and updated my `environment.yml` so that is is the dame to the file in the `dev` branch - I tend to miss up with `git`, I am sorry, can I do something to solve this?

drpatelh commented:
Git gets me sometimes too 👍 There may be a more cleverer way to do this but for a small change like this I would just re-download all 3 original files you have amended and then make the changes again. Git should work out the differences based on that. e.g.

`wget -L https://raw.githubusercontent.com/nf-core/rnaseq/dev/environment.yml`
`wget -L https://raw.githubusercontent.com/nf-core/rnaseq/dev/main.nf`
`wget -L https://raw.githubusercontent.com/nf-core/rnaseq/dev/CHANGELOG.md`

silviamorins commented:
I should have fixed it .. ?

apeltzer commented:
environment.yml missing - that should be required too 👍

silviamorins commented:
But I haven't changed it, is the same as in `https://raw.githubusercontent.com/nf-core/rnaseq/dev/environment.yml`

apeltzer commented:
Hmm, the PR says you did ;-) 

https://github.com/nf-core/rnaseq/pull/248/files

silviamorins commented:
I removed it and then downloaded it again with `wget -L https://raw.githubusercontent.com/nf-core/rnaseq/dev/environment.yml`. So it's the same as in the `dev` branch now.

drpatelh commented:
When did you fork the pipeline? Im wondering if its because you havent updated the code on your local branch...

drpatelh commented:
Can you try running these commands:
`git remote add upstream https://github.com/nf-core/rnaseq.git`
`git pull upstream dev`                                        
`git push`

silviamorins commented:
I did as you said, had a conflicht on the `CHANGELOG`, solved it and pushed. It still doesn't look alright to me ..

drpatelh commented:
Looks fine now 👍 Only the files you amended are showing up now with only the changes you made.

silviamorins commented:
Great, thanks! sorry for the loss of time. Will try not to mess it up next time.

drpatelh commented:
No worries. The trick is to make sure you update the code on your fork every time you want to add new commits because this will factor in new changes to `dev` since your last changes.
